tlslite/tlsrecordlayer.py

```diff
@@ -722,6 +725,31 @@ def _getMsg(self, expectedType, secondaryType=None, constructorType=None):
                                 yield result
                             continue
 
+                    # If we received a heartbeat request ...
+                    if recordHeader.type == ContentType.heartbeat:
+                        # If heartbeat extension was not negotiated, then
+                        # we send unexpected_message fatal alert
+                        if not self.heartbeat:
+                            for result in self._sendError( \
+                                    AlertDescription.unexpected_message,
+                                            "Heartbeat was not negotiated"):
+                                yield result
+
+                        try:
+                            heartbeat_message = Heartbeat().parse(p)
```

Comment:
At `_getMsg` is `try:`, that is catching exceptions raised by a Parser or Message instances and sending `decode_error` alert. Or we want something different, when we fail at parsing heartbeat message?